python/examples/dspy-rag-fastapi/backend/app/utils/load.py

```diff
@@ -23,74 +24,58 @@ def chunks(iterable, chunk_size):
-            input (Documents): A list of texts to get embeddings for.
-        Returns:
-            Embeddings: The embeddings for the texts.
-        Example:
-            >>> openai = OpenAIEmbeddingFunction()
-            >>> texts = ["Hello, world!", "How are you?"]
-            >>> embeddings = openai(texts)
-        """
-
-        embeddings: Embeddings = []
-        # Call OpenAI Embedding API for each document.
-        for document_chunk in chunks(input, chunk_size=10):
-            embedding_response = self._client.embeddings.create(
-                model=self._model_name, input=document_chunk
-            )
-            embeddings.extend([emb.embedding for emb in embedding_response.data])
-
-        return embeddings
-
-
 def load_data() -> None:
     """
-    Loads data from /data/example to Chroma Vector store.
+    Loads data from /data/example to a Weaviate Vector store.
     """
 
     logger.info("Loading data.")
     # Split document into single sentences
-    chunks = []
+    sentence_chunks = []
     with open(f"{DATA_DIR}/example/paul_graham_essay.txt", "r", encoding="utf-8") as file:
         text = file.read()
         sentences = re.split(r"(?<!\w\.\w.)(?<![A-Z][a-z]\.)(?<=\.|\?)\s", text)
         sentences = [sentence.strip() for sentence in sentences if sentence.strip()]
-        chunks.extend(sentences)
+        sentence_chunks.extend(sentences)
 
     logger.info("Creating embeddings.")
-    openai_ef = OpenAIEmbeddingFunction()
-    chunks_embeddings = openai_ef(chunks)
-
-    db = chromadb.PersistentClient(path=f"{DATA_DIR}/chroma_db")
-    chroma_collection = db.get_or_create_collection("quickstart")
 
-    logger.info("Loading data in Chroma.")
-    chroma_collection.add(
-        ids=[f"id{i}" for i in range(1, len(chunks) + 1)],
-        embeddings=chunks_embeddings,
-        documents=chunks,
+    weaviate_client = weaviate.WeaviateClient(
+        embedded_options=weaviate.embedded.EmbeddedOptions(
+            persistence_data_path=f"{DATA_DIR}/weaviate",
+            additional_env_vars={
+                "ENABLE_MODULES": "text2vec-openai",
+                "DEFAULT_VECTORIZER_MODULE": "text2vec-openai",
+            },
+        ),
+        additional_headers={"X-OpenAI-Api-Key": os.getenv("OPENAI_API_KEY")},
     )
-    logger.info("Successfully loaded embeddings in the Chroma.")
+    weaviate_client.connect()
+
+    try:
+        openai_client = openai.Client()
+
+        weaviate_client.collections.delete("paul_graham_essay")
+        collection = weaviate_client.collections.create(
+            "paul_graham_essay", vectorizer_config=wvc.config.Configure.Vectorizer.text2vec_openai()
+        )
+
+        logger.info("Loading data into Weaviate.")
+        with collection.batch.dynamic() as batch:
+            for sentence_chunk in chunks(sentence_chunks, 10):
+                embedding_response = openai_client.embeddings.create(
+                    model="text-embedding-ada-002", input=sentence_chunk
```

Comment:
Not sure if it matters, but this is not the latest embedding model.

Reply:
yeah weaviate defaults to this. It's a funky setup where the embeddings happen on the server.